tests/UnitTests/rwlock_ut/rwlock.cpp

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <utcommon.h>
+
+#include <k4ainternal/rwlock.h>
+#include <k4ainternal/common.h>
+#include <gtest/gtest.h>
+
+#include <azure_c_shared_utility/lock.h>
+#include <azure_c_shared_utility/tickcounter.h>
+#include <azure_c_shared_utility/threadapi.h>
+
+int main(int argc, char **argv)
+{
+    return k4a_test_common_main(argc, argv);
+}
+
+#define GTEST_LOG_INFO std::cout << "[     INFO ] "
+
+TEST(rwlock_ut, rwlock_init)
+{
+    k4a_rwlock_t lock;
+
+    rwlock_init(&lock);
+
+    rwlock_deinit(&lock);
+}
+
+typedef struct
+{
+    k4a_rwlock_t test_lock;
+
+    // Number of times reader 1 gets the lock
+    volatile int reader1_count;
+
+    // Number of times reader 2 gets the lock
+    volatile int reader2_count;
+
+    // Number of times reader 2 fails to get the lock
+    volatile int reader2_fail_count;
+
+    // Number of times writer 1 gets the lock
+    volatile int writer1_count;
+
+    // Number of times writer 2 gets the lock
+    volatile int writer2_count;
+
+    // Number of times writer 2 fails to get the lock
+    volatile int writer2_fail_count;
+
+    volatile bool run_test;
+} threaded_rwlock_test_context_t;
+
+static void thread_reader1(threaded_rwlock_test_context_t *context)
+{
+    while (context->run_test)
+    {
+        rwlock_acquire_read(&context->test_lock);
+
+        int writer1 = context->writer1_count;
+        int writer2 = context->writer2_count;
+
+        context->reader1_count++;
+
```

Comment:
Would you want a sleep here to make sure that both readers can be going in at the same time, and to make sure that the writers will block on this reader being outstanding?